spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/DatastoreOperations.java

```diff
@@ -0,0 +1,114 @@
+/*
+ *  Copyright 2018 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.data.datastore.core;
+
+import java.util.List;
+
+import com.google.cloud.datastore.Entity;
+import com.google.cloud.datastore.Key;
+import com.google.cloud.datastore.Query;
+
+import org.springframework.lang.Nullable;
+
+/**
+ * An interface of operations that can be done with Cloud Datastore.
+ *
+ * @author Chengyuan Zhao
+ *
+ * @since 1.1
+ */
+public interface DatastoreOperations {
+
+	/**
+	 * Get an entity based on a key.
+	 * @param entityClass the type of the entity to get.
+	 * @param key the key of the entity
+	 * @param <T> the class type of the entity.
+	 * @param readOptions optional read options.
+	 * @return the entity that was found with that key.
+	 */
+	<T> T read(Class<T> entityClass, Key key, @Nullable DatastoreReadOptions readOptions);
```

Comment:
Looking at `MongoTemplate` and [Spring Data JDBC](https://github.com/spring-projects/spring-data-jdbc/blob/master/src/main/java/org/springframework/data/jdbc/core/JdbcAggregateOperations.java) and some other Spring Data examples, `findById` seems more common.  Why are we straying from that pattern?
Also in the examples I've seen the key is passed in as an `Object`. I'm not sure why though.